wrap_test.go

```diff
@@ -89,18 +93,22 @@ func BenchmarkDecodeBlockParallel(b *testing.B) {
 	b.ResetTimer()
 	var wg sync.WaitGroup
 	wg.Add(100)
+	errs := make(chan error, 100)
 	for j := 0; j < 100; j++ {
 		go func() {
 			defer wg.Done()
 			for i := 0; i < b.N; i++ {
-				nd2, err := DecodeBlock(nd)
-				if err != nil {
-					b.Fatal(err, nd2)
+				if _, err := DecodeBlock(nd); err != nil {
+					errs <- err
 				}
 			}
 		}()
 	}
 	wg.Wait()
+	close(errs)
+	for e := range errs {
+		b.Fatal(e)
```

Comment:
Note: you can also do this by calling `b.Error` inside the goroutine, then waiting. We don't actually _need_ Fatal here because we don't need to exit early.